File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer,Integer> hm=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int num=nums[i];
            if(hm.containsKey(target-num)){
                return new int[]{hm.get(target-num),i};
            }
            hm.put(num,i);}
            return new int[]{};
    }
        }

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hm;
        for (int i=0;i<nums.size();i++){
            int num=nums[i];
            if (hm.find(target-num)!=hm.end()) {
                return{hm[target-num],i};
            }
            hm[num]=i;
        }
        return {}; 
    }
};
